Sources/App/Core/Dependencies/CurrentReferenceCacheClient.swift

```diff
@@ -0,0 +1,86 @@
+// Copyright Dave Verwer, Sven A. Schmidt, and other contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Dependencies
+import DependenciesMacros
+
+
+@DependencyClient
+struct CurrentReferenceCacheClient {
+    var set: @Sendable (_ owner: String, _ repository: String, _ reference: String?) async -> Void
+    var get: @Sendable (_ owner: String, _ repository: String) async -> String?
+}
+
+
+extension CurrentReferenceCacheClient: DependencyKey {
+    static let timeToLive: Duration = .seconds(5*60)
+
+    static var liveValue: CurrentReferenceCacheClient {
+        .init(
+            set: { owner, repository, reference async in
+                @Dependency(\.redis) var redis
+                await redis.set(key: getKey(owner: owner, repository: repository),
+                                value: reference,
+                                expiresIn: timeToLive)
```

Comment:
@heckj , this is how you'd interface with Redis once this lands. We wouldn't be using `app.redis` at all but rather the private `Redis` actor via the `RedisClient` dependency, replacing your PR #3580.

The only thing this doesn't cover is setup for local use of Redis, i.e. during development. It should be easy to add though, either by making the hostname configurable or just setting a `redis` alias in `/etc/hosts` on the local machine. The latter is certainly easiest and fastest to get going.

Reply:
Correction, I just realised I misread your PR #3580. I was focused on the part

```diff
-        app.redis.configuration = try RedisConfiguration(hostname: "redis")
+        if let redisHost = Environment.get("REDIS_HOST") {
+            app.redis.configuration = try RedisConfiguration(hostname: redisHost)
+        } else {
+            app.logger.warning("REDIS_HOST not set, caching disabled.")
+        }
```

but really the point of it is to add a bringup for a local Redis and the hostname config.